v2/emailpassword/troubleshooting/cors-issues.mdx

```diff
@@ -0,0 +1,219 @@
+
+This is important because Supertokens has both a frontend and backend SDK. The frontend SDK calls the APIs exposed by the backend SDK. As a result, if the origin to set to false (disables CORS) or the wrong origin, we’ll find an error such as the following -
+
+```
+Origin [http://localhost:3000](http://localhost:3000/) is not allowed by Access-Control-Allow-Origin. Status code: 204
+```
+
+### `allowedHeaders` [optional]
+
+This configures the `Access-Control-Allow-Headers` CORS header. It expects one of two input types.
+
+- Comma-delimited string - `'X-Custom-Header, Content-Type'`
+- Array - `['X-Custom-Header', 'Content-Type']`
+
+For Supertokens projects, we can use the SuperTokens.getCORSAllowedHeaders() function which returns the following list - `["anti-csrf", "rid"]`
+
+This is important because the header tells the browser what HTTP headers are allowed to be used when making a cross-origin request.
+
+### `methods` [optional]
+
+This configures the Access-Control-Allow-Methods CORS header. It expects one of two input types.
+
+- Comma-delimited string - `'GET,PUT,POST'`
+- Array - `['GET', 'PUT', 'POST']`
+
+This is important because the header tells the browser what HTTP methods are allowed to be used when making a cross-origin request. Supertokens in particular uses `GET`, `POST` and `PUT` for functions such as creating new accounts, signing in/out etc.
+
+### `credentials` [must have]
+
+This configures the `Access-Control-Allow-Credentials` CORS header. It expects a boolean.
+
+If you set this header to true, then the browser will allow cookies to be sent in cross-origin requests. This is important because SuperTokens, by default uses cookies for sessions in the browser.
+
+## Common CORS Issues
+<br/>
+
+### `[Error] Credentials flag is true, but Access-Control-Allow-Credentials is not "true".`
+
+This happens when the CORS credentials field is not set to true.
+
+We can remedy this issue by setting credentials to `true` in our cors setting.
+
+```tsx
+import SuperTokens from "supertokens-node";
+import express from "express";
+import cors from "cors";
+let app = express();
+
+// ...other middlewares
+
+app.use(
+  cors({
+    origin: "http://127.0.0.1:3000",
+    allowedHeaders: ["content-type", ...SuperTokens.getAllCORSHeaders()],
+    methods: ["GET", "PUT", "POST", "DELETE"],
+    //highlight-start
+    credentials: true, // Check this line
+    //highlight-end
+
+  })  
+);
+```
+
+<br/>
+
+### `[Error] Origin [http://localhost:3000](http://localhost:3000/) is not allowed by Access-Control-Allow-Origin. Status code: 204`
+
+This happens when the CORS origin field is incorrect. Another varient of this error is when the OPTIONS API returns a 404 error. This can happen cause of misconfigured API gateway or reverse proxy. It can also be caused by a misconfiguration in the CORS origin field. 
```

Comment:
This is talking about two issues in one paragraph in a jumbled way.. please rephrase (one issue about incorrect origin, and one about misconfig of proxy).

Reply:
Ok sounds good!